editor/src/core/model/performance-scripts.tsx

```diff
@@ -204,19 +217,75 @@ export function useTriggerSelectionPerformanceTest(): () => void {
     'useTriggerSelectionPerformanceTest dispatch',
   )
   const allPaths = useRefEditorState((store) => store.derived.navigatorTargets)
+  const selectedViews = useRefEditorState((store) => store.editor.selectedViews)
   const trigger = React.useCallback(async () => {
+    const targetPath = [...allPaths.current].sort(
+      (a, b) => EP.toString(b).length - EP.toString(a).length,
+    )[0]
+    // Determine where the events should be fired.
+    const controlsContainerElement = forceNotNull(
+      'Container controls element should exist.',
+      document.getElementById(CanvasControlsContainerID),
+    )
+
+    const targetElement = forceNotNull(
+      'Target element should exist.',
+      document.querySelector(`*[data-paths~="${EP.toString(targetPath)}"]`),
+    )
+    const targetBounds = targetElement.getBoundingClientRect()
+    await dispatch([CanvasActions.positionCanvas(canvasPoint({ x: -1900, y: -1100 }))], 'everyone')
+      .entireUpdateFinished
     if (allPaths.current.length === 0) {
       console.info('SELECT_TEST_ERROR')
       return
     }
 
-    const targetPath = [...allPaths.current].sort(
-      (a, b) => EP.toString(b).length - EP.toString(a).length,
-    )[0]
     let framesPassed = 0
     async function step() {
       performance.mark(`select_step_${framesPassed}`)
-      await dispatch([selectComponents([targetPath!], false)]).entireUpdateFinished
+      controlsContainerElement.dispatchEvent(
+        new MouseEvent('mousedown', {
+          detail: 1,
+          bubbles: true,
+          cancelable: true,
+          metaKey: true,
+          clientX: targetBounds.left + 5,
+          clientY: targetBounds.top + 5,
+          buttons: 1,
+        }),
+      )
+      function isTargetSelected(): boolean {
+        return ElementPathArrayKeepDeepEquality([targetPath], selectedViews.current).areEqual
+      }
+      const startingTime = Date.now()
+      while (!isTargetSelected() && Date.now() < startingTime + 3000) {
+        await wait(50)
+      }
+      if (!isTargetSelected()) {
+        throw new Error(`Element never ended up being selected.`)
+      }
+      controlsContainerElement.dispatchEvent(
+        new MouseEvent('pointerup', {
+          detail: 1,
+          bubbles: true,
+          cancelable: true,
+          metaKey: true,
+          clientX: targetBounds.left + 5,
+          clientY: targetBounds.top + 5,
+          buttons: 1,
+        }),
+      )
+      controlsContainerElement.dispatchEvent(
+        new MouseEvent('mouseup', {
+          detail: 1,
+          bubbles: true,
+          cancelable: true,
+          metaKey: true,
+          clientX: targetBounds.left + 5,
+          clientY: targetBounds.top + 5,
+          buttons: 1,
+        }),
+      )
       performance.mark(`select_dispatch_finished_${framesPassed}`)
```

Comment:
at this point, did everything caused by the mouse event dispatch finish synchronously? shouldn't we be awaiting for the dom walker here?

Reply:
According to the profile I did previously the logic for these fires synchronously with the mouse events.